aw_datastore/storages/sqlite.py

```diff
@@ -102,10 +108,13 @@ def conditional_commit(self, num_statements):
         This is because sqlite is very slow with small inserts, this
         is a way to batch them together and lower CPU+disk usage
         """
-        self.num_uncommited_statements += num_statements
-        if self.num_uncommited_statements > 50:
-            self.commit()
-        if (self.last_commit - datetime.now()) > timedelta(seconds=10):
+        if self.enable_lazy_commit:
+            self.num_uncommited_statements += num_statements
+            if self.num_uncommited_statements > 10:
```

Comment:
Why lower it from 50 to 10?

And why do you need the enable_lazy_commit at all? It should work transparently?

Reply:
Sorry, the 50 to 10 was accidental. Fixed.

`enable_lazy_commit` effectively makes the code asynchronous, which wasn't acceptable for a use-case I had (sync).

Reply:
I understood that the use-case was sync, but why do you really need it?

Why do you need the absolute latest event? If you actually do, can't you instead force a commit when you actually need it like when using the REST API?

Reply:
>Why do you need the absolute latest event?

Because I'm checking for the latest sent event sometimes, to know which events to send (I'd need to know about eventual uncommitted events).

>If you actually do, can't you instead force a commit when you actually need it like when using the REST API?

I am calling it through the ServerAPI wrapper in `aw_server/api.py`, and there doesn't seem to be any commit done through the REST API? How would you force a commit?

Reply:
https://github.com/ActivityWatch/aw-server/blob/master/aw_server/api.py#L53
https://github.com/ActivityWatch/aw-core/blob/master/aw_datastore/storages/sqlite.py#L224

If that's actually what you are doing you should be getting the latest event.

Reply:
Well, that's weird, because I definitely didn't (That's why I did this in the first place). I'll take a closer look at it.

Reply:
Maybe it's because you are running it as a seperate process?

Same thing here as with aw-server-rust, you should not be accessing a SQLite database from multiple threads or processes.

Reply:
>Maybe it's because you are running it as a seperate process?

I'm not, I'm instantiating two `ServerAPI` objects with two different `SqliteDatastore` objects (with different db paths) in the same process. 

I've reproduced the issue by setting `enable_lazy_commit=True`, not sure what is going wrong, only seems to happen if process dies shortly after commit?

Reply:
I can't figure it out, I'm going ahead and merging this anyway.